### PR TITLE
build: remove sphinx-book-theme constraint to use latest release

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR removes the existing `sphinx-book-theme` to use the latest release. Here's the comment we realized we need to utilize the constraint: https://github.com/openedx/openedx-events/pull/169#discussion_r1073980224
Also, we updated the theme configurations for the latest suggested on the open edx docs: https://docs.openedx.org/en/latest/developers/how-tos/add-sphinx-docs-to-a-repo.html.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: requirements/doc.txt [hunk 1]
```diff
@@ -197,10 +201,8 @@ sphinx==5.3.0
     #   sphinxcontrib-contentui
 sphinx-autobuild==2021.3.14
     # via -r requirements/doc.in
-sphinx-book-theme==0.4.0rc1
-    # via
-    #   -c requirements/constraints.txt
-    #   -r requirements/doc.in
+sphinx-book-theme==1.0.1
```
Comment (felipemontoya) [inline requirements/doc.txt:204]: they went all the way to stable v1 and then at least one patch. Great!

**Review by felipemontoya — APPROVED** [review]
Looks great. One constraint down

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-events&entry.867055334=https://github.com/openedx/openedx-events/pull/306&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2024-01-24+16:17&entry.1917517419=2024-01-24+17:18&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**